metabase/metabase
# Fix model level persistence UI states

Related to: #22538

We're going to ship model-level persistence in paid plans. We have some prototype code turning on/off persistence on a modal + a dedicated section in the details sidebar. The PR fixes a few things:

1. Makes sure we don't show the "Persist/unpersist modal" button and the sidebar cache info section in OSS
2. Fixes you had to refresh the page to see how UI changes after persistence is turned on/off
3. Fixes it was feeling weird when you clicked the "Persist/unpersist" button. The icon for on/off state are the same (it's a prototype and it will be changed) and it wasn't clear if something actually happened when you clicked the button.

### To Verify

1. Run Metabase OSS
2. Go to `/admin/settings/caching` and enable model persistence
3. Spin up a sample PostgreSQL DB from [metabase/metabase-qa](https://github.com/metabase/metabase-qa) in Docker
4. Go to `/admin/databases`, connect the new PSQL DB and enable model persistence for the DB (click the "Enable model persistence" button on the DB page's sidebar)
5. Create a question using PSQL data, turn it into a model
6. Open model details sidebar (click the title)
7. Ensure you can't see the "Enable persistence" button ("database" icon) in the actions panel
8. Run Metabase Enterprise
9. Visit the same model and make sure you now see the "Enable persistence" button
